docs: add NestHybridApplicationOptions documentation (#1112)
nestjs/docs.nestjs.com: content/faq/hybrid-application.md

await app.startAllMicroservicesAsync();
await app.listen(3001);
```

#### Sharing configuration

> By default hybrid application will not inherit global pipes, interceptors, guards & filters.

Reviewer: ```suggestion
By default a hybrid application will not inherit global pipes, interceptors, guards & filters configured for the main (HTTP-based) application.
```

> By default a hybrid application will not inherit global pipes, interceptors, guards & filters configured for the main (HTTP-based) application.

Reviewer: ```suggestion
By default a hybrid application will not inherit global pipes, interceptors, guards and filters configured for the main (HTTP-based) application.
```

> You'll need to set inheritAppConfig in the second argument as follow:

Reviewer: ```suggestion
To inherit these properties from the main application, set the `inheritAppConfig` property in the second argument (an optional options object) of the `connectMicroservice()` call, as follow:
```

> To inherit these properties from the main application, set the `inheritAppConfig` property in the second argument (an optional options object) of the `connectMicroservice()` call, as follow:

Reviewer: ```suggestion
To inherit these configuration properties from the main application, set the `inheritAppConfig` property in the second argument (an optional options object) of the `connectMicroservice()` call, as follow:
```
